src/selectors.js

```diff
@@ -2,8 +2,16 @@ import { matchPath } from "react-router"
 
 const createSelectors = (structure) => {
   const { getIn, toJS } = structure
-  const getLocation = state => toJS(getIn(state, ['router', 'location']))
-  const getAction = state => toJS(getIn(state, ['router', 'action']))
+  const _getLocation = state => getIn(state, ['location'])
+  const _getAction = state => getIn(state, ['action'])
+  const isRouter = (value) => value != null && typeof value === 'object' && _getLocation(value) && _getAction(value)
+  const getRouter = state => {
+    const router = toJS(getIn(state, ['router']))
+    if (!isRouter(router)) { throw 'Could not find router reducer in state tree, it must be mounted under "router"' }
+    return router
+  }
+  const getLocation = state => toJS(_getLocation(getRouter(state)))
+  const getAction = state => toJS(_getAction(getRouter(state)))
```

Comment:
Seems weird to me to be exporting these selectors always wrapped in toJS, is this something we could move somewhere immutable/seamless-immutable specific?

Reply:
I prefer keeping it here for backward compatibility. By the way, `_getLocation` under `getLocation` seems redundant to me. `getIn(getRouter(state), ['location'])` seems to be more straightforward.

Reply:
Yeah that's fine, was just trying to eliminate duplication but I'll change it